src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1284,18 +1284,25 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
                     null, Metadata.ValidateCountNotInArray);
             }
 
+            if (MinLength == MaxLength && len != MaxLength)
```

Comment:
The fully qualified error ids on parameter validation errors aren't too interesting, I'm not worried about changing those.

Personal opinion - adding prescriptive guidance may have been the cause for the original bug - having a confusing error message. It's easier to get technically accurate errors - what is wrong - than it is to suggest how to fix the problem, so it can be better to not bother. And in this specific case, I think it's fairly obvious how to fix the error.